test/extended/bootstrap_user/bootstrap_user_login.go

```diff
@@ -0,0 +1,31 @@
+package bootstrap_user
+
+import (
+	"io/ioutil"
+	"path/filepath"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+var _ = g.Describe("[Conformance] The bootstrap user", func() {
+	defer g.GinkgoRecover()
```

Comment:
@smarterclayton It's unclear to me what the exact label for this should be.  `Conformance` ok?

Reply:
Depends on what this is supposed to test.  You can't couple an extended test to the installer artifacts (e2e tests have to run against anyone's cluster).  Mo owns testing that we can set / update a bootstrap user on a cluster, so at that point you're more about establishing that a newly created cluster has a bootstrap user.  However, not all users are going to run extended against a cluster with a bootstrap user, so we'd have to make the test conditional on that behavior.

* A test that verifies the installer itself creates a bootstrap user would belong to the installer and be in that repo.
* A test that verifies that a conformant openshift cluster can have a bootstrap user and exercises the secret creation paths and verifies oauth logins work would be here, but would not be part of the conformance suite (this is what I alluded to mo owning)

I think you're trying to build more of the former rather than the latter - since you're trying to verify that the installer correctly sets up the password AND that the password is useful, so this really belongs in the installer test suite.

Reply:
We also want to confirm that what has to happen via authentication/osin/oauthconfig for token-generation is not broken.  
We could go back to my original plan of checking that both are kosher in the cluster-launch-installer-* templates, by logging in/switching back to system:admin context.

Reply:
> verify that the installer correctly sets up the password AND that the password is useful
* the installer already 'tests' that the password is created, by spitting it out at the end (and errors if not read, in cmd/openshift-install/create.go).  
* to test that it's useful, have to go through oauthserver. I still don't see how that test belongs in the installer. It requires an unsupportedConfigOverrides oauthConfig section in kubeAPIServerOperatorConfig patched, etc.  Only way to test that that completed is to run `oc login` with kubeadmin
.   
> this really belongs in the installer test suite.

AFAIK there isn't an installer test suite atm? There _was_ a smoke tests suite but it's not currently run/they're outdated/tectonic references still.  I can look at updating those, but that might take awhile, and the auth team needs assurance now that upcoming changes don't break kube:admin. 

> A test that verifies that a conformant openshift cluster can have a bootstrap user and exercises the secret creation paths and verifies oauth logins work would be here

Yes, this!  So I need to make this extended test conditional on clusters that have the bootstrap kubeadmin password (ie, any cluster created with openshift/installer)?  Or, easiest way (for now) is to put this check in the cluster-launch-installer-* CI templates, for all jobs running against a cluster launched w/ openshift/installer like so: https://github.com/openshift/release/pull/2440

Reply:
@enj mentioned I can create the kubeadmin secret in the extended test that conforms to the pw requirements, if that pw doesn't already exist, then check for kubeadmin user login, since every 4.0 cluster has the ability to create kubeadmin user, I'll go with that.